lessons/en/chapter_7.yaml

```
@@ -38,6 +38,39 @@
 
     That said, Rust implements many programming language features, so that you
     might not mind this lacking.
+- title: self and Self
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=struct+Coordinates+%7B%0A++++x%3A+i32%2C%0A++++y%3A+i32%2C%0A%7D%0A%0Aimpl+Coordinates+%7B%0A++++%2F%2F+the+return+value%2C+Self%2C+has+the+data+type+as+the+implemented+struct+%0A++++%2F%2F+this+method+is+static%0A++++fn+new%28xval%3A+i32%2C+yval%3A+i32%29+-%3E+Self+%7B%0A++++++++return+Coordinates+%7B%0A++++++++++++x%3A+xval%2C%0A++++++++++++y%3A+yval%2C%0A++++++++%7D%0A++++%7D%0A++++%0A++++fn+setx%28%26mut+self%2C+xval%3A+i32%29+%7B%0A++++++++self.x+%3D+xval%3B%0A++++%7D%0A++++%0A++++fn+sety%28%26mut+self%2C+yval%3A+i32%29+%7B%0A++++++++self.y+%3D+yval%3B%0A++++%7D%0A++++%0A++++fn+disp%28%26self%29+%7B%0A++++++++println%21%28%22Point+located+at+%3A+%7B%7D+on+OX++%7B%7D+on+OY%22%2C%0A++++++++++++++++self.x%2C+self.y%29%3B%0A++++%7D%0A++++%0A%7D%0A%0Afn+main%28%29+%7B%0A++++let+mut+p+%3D+Coordinates%3A%3Anew%28-1i32%2C+1i32%29%3B%0A++++p.disp%28%29%3B%0A++++p.setx%2810++as+i32%29%3B%0A++++p.sety%28-10+as+i32%29%3B%0A++++p.disp%28%29%3B%0A%7D%0A
+  content_markdown: >
+      In Rust, `self` and `Self` are fundamental concepts in ownership and
+      borrowing system. Both of them are used in `impl` and `trait` blocks.
+      
+      `self`:
+      - refers to an **instance of struct**
+      - reference to itself
+      - is the first parameter of the method
+      - methods containing the `self` parameter are called using the name of the instance, the operator `.` and the name of the method
+    
+      
+      `Self`:
+      - refers to **return type** of the method
+      - has the same data type as the `struct` that is implemented in a `impl` block
+      - can be used for static method using the operator `::` (static method do not depend on a instance of a struct and a static method does not contain `self` as parameter)
```

"static methods" appears too many times in the same phrase. This can be avoided:
```suggestion
      - can be used for static methods using the operator `::` (they do not depend on a instance of a struct or contain `self` as a parameter)
```